kratos/includes/kernel.h

```diff
@@ -151,31 +113,21 @@ class KRATOS_API(KRATOS_CORE) Kernel
         unsigned int j = 0;
         for(KratosComponents<VariableData>::ComponentsContainerType::iterator i = KratosComponents<VariableData>::GetComponents().begin() ;
                 i != KratosComponents<VariableData>::GetComponents().end() ; i++)
-            //const_cast<VariableData&>(i->second.get()).SetKey(++j);
             i->second->SetKey(++j);
     }
 
     /// Initializes and synchronizes the list of variables, elements and conditions in each application.
     /** This method gives the application the list of all variables, elements and condition which is registered
         by kratos and all other added applications.
-
-
         @see AddApplication
         @see Initialize
     */
     void InitializeApplication(KratosApplication& NewApplication)
     {
-        /*
-		NewApplication.SetComponents(KratosComponents<VariableData>::GetComponents());
-        NewApplication.SetComponents(KratosComponents<Element>::GetComponents());
-        NewApplication.SetComponents(KratosComponents<Condition>::GetComponents());
-
-        NewApplication.GetRegisteredObjects().insert(Serializer::GetRegisteredObjects().begin(), Serializer::GetRegisteredObjects().end());
-        NewApplication.GetRegisteredObjectsName().insert(Serializer::GetRegisteredObjectsName().begin(), Serializer::GetRegisteredObjectsName().end());
-		*/
     }
 
-
+    bool HasApplication(std::string ApplicationName);
```

Comment:
I was thinking a bit more about this, and I do not like the `HasApplication` syntax. What would you think of `IsImported("ApplicationName")`?

Reply:
1line change... no problem for me

Reply:
@jcotela, done!